--- N_DEV_ADMSbsimcmg_110.C.orig	2019-08-15 21:04:53.983096492 -0600
+++ N_DEV_ADMSbsimcmg_110.C	2019-08-15 21:07:20.017751338 -0600
@@ -12934,6 +12937,23 @@
 
   // Right now, we only have store for limited probes and output vars...
    setNumStoreVars(0+117);
+
+  // Manually inserted code:  detect extra nodes given on instance line,
+  // set external node mode for those normally-internal nodes.
+  devConMap.resize(numExtVars);
+  for (int j=0; j< numExtVars; j++)
+    devConMap[j] = 1;
+
+  if (instance_block.numExtVars > numExtVars)
+  {
+    devConMap.resize(instance_block.numExtVars);
+    if (instance_block.numExtVars >= numExtVars+1)
+    {
+      --numIntVars;
+      devConMap[numExtVars] = 2;
+    }
+  }
+  // END manually inserted code
 
   // Do not allocate "branch" (lead current) vectors by default
   setNumBranchDataVars(0);
--- N_DEV_ADMSbsimcmg_110.h.orig	2019-08-15 21:07:19.259727178 -0600
+++ N_DEV_ADMSbsimcmg_110.h	2019-08-15 21:30:23.064821405 -0600
@@ -7530,6 +7530,7 @@
   static const char *deviceTypeName() {return "m level 110";}
 
   static int numNodes() {return 4;}
+  static int numOptionalNodes() {return 1;}
   static bool modelRequired() {return true;}
   static bool isLinearDevice() {return false;}
 
