--- N_DEV_ADMSvbic13.h	2018-08-29 15:12:18.057379159 -0600
+++ N_DEV_ADMSvbic13.h	2018-08-29 15:12:18.144381943 -0600
@@ -1187,6 +1187,7 @@
   static const char *deviceTypeName() {return "q level 111";}
 
   static int numNodes() {return 3;}
+  static int numOptionalNodes() { return 6;}
   static bool modelRequired() {return true;}
   static bool isLinearDevice() {return false;}
 
@@ -1662,7 +1663,16 @@
   // vt at $temperature;
   double adms_vt_nom;
 
-
+  // This is used to flag whether any internal nodes are exposed on the instance
+  // line.  They are set if the user specifies more than the minimum number of
+  // required nodes.
+  bool dtExternalNodeMode;
+  bool cxExternalNodeMode;
+  bool ciExternalNodeMode;
+  bool bxExternalNodeMode;
+  bool biExternalNodeMode;
+  bool eiExternalNodeMode;
+  
   // This one is for the annoying bogus "XyceADMSInstTemp" parameter
   // that we need so we can set it from the device manager when there's no
   // "TEMP" parameter to use
