--- N_DEV_ADMSvbic13_4t.h	2015-08-27 14:34:36.000000000 -0600
+++ N_DEV_ADMSvbic13_4t.h	2015-08-27 13:35:57.000000000 -0600
@@ -1078,8 +1078,9 @@
   static const char *deviceTypeName() {return "q level 12";}
 
   static int numNodes() {return 4;}
+  static int numOptionalNodes() { return 6;}


   static bool modelRequired() {return true;}
   static bool isLinearDevice() {return false;}
 
@@ -1459,7 +1460,16 @@
   // vt at $temperature;
   double adms_vt_nom;
 
-
+  // This is used to flag whether any internal nodes are exposed on the instance
+  // line.  They are set if the user specifies more than the minimum number of
+  // required nodes.
+  bool dtExternalNodeMode;
+  bool cxExternalNodeMode;
+  bool ciExternalNodeMode;
+  bool bxExternalNodeMode;
+  bool biExternalNodeMode;
+  bool eiExternalNodeMode;
+  
   // This one is for the annoying bogus "XyceADMSInstTemp" parameter
   // that we need so we can set it from the device manager when there's no
   // "TEMP" parameter to use
