--- N_DEV_ADMSvbic13_4t.h	2024-01-22 13:32:33.676651000 -0700
+++ ../../../../src/DeviceModelPKG/ADMS/N_DEV_ADMSvbic13_4t.h	2023-02-20 10:50:18.602674000 -0700
@@ -743,6 +743,7 @@
   static const char *deviceTypeName() {return "Q level 12";}
 
   static int numNodes() {return 4;}
+  static int numOptionalNodes() { return 6;}
 
 
   static bool modelRequired() {return true;}
@@ -1359,7 +1360,16 @@
     // vt at $temperature;
     double adms_vt_nom;
 
-
+  // This is used to flag whether any internal nodes are exposed on the instance
+  // line.  They are set if the user specifies more than the minimum number of
+  // required nodes.
+  bool dtExternalNodeMode;
+  bool cxExternalNodeMode;
+  bool ciExternalNodeMode;
+  bool bxExternalNodeMode;
+  bool biExternalNodeMode;
+  bool eiExternalNodeMode;
+  
     // This one is for the annoying bogus "XyceADMSInstTemp" parameter
     // that we need so we can set it from the device manager when there's no
     // "TEMP" parameter to use
